sdk/eventhubs/azure-messaging-eventhubs/README.md

```diff
@@ -1,3 +1,186 @@
+
+Events can be consumed from an event hub using an [event consumer](https://docs.microsoft.com/azure/event-hubs/event-hubs-features#event-consumers). In this package there are two types for consuming events: 
+- The basic event consumer is the PartitionClient, in the [ConsumerClient](https://azure.github.io/azure-sdk-for-cpp/storage.html). This consumer is useful if you already known which partitions you want to receive from.
+- A distributed event consumer, which uses Azure Blobs for checkpointing and coordination. This is implemented in the [Processor](https://azure.github.io/azure-sdk-for-cpp/storage.html). 
+The Processor is useful when you want to have the partition assignment be dynamically chosen, and balanced with other Processor instances.
+
+More information about Event Hubs features and terminology can be found here: [link](https://docs.microsoft.com/azure/event-hubs/event-hubs-features)
+
+
+# Examples
+
+<!-- NOTE: Fix dead links -->
+Examples for various scenarios can be found on [azure.github.io](https://azure.github.io/azure-sdk-for-cpp/storage.html) or in the samples directory in our GitHub repo for 
+[EventHubs](https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/eventhubs).
+
+## Send events
+
+The following example shows how to send events to an event hub:
+
+```cpp
+// Your Event Hubs namespace connection string is available in the Azure portal.
+std::string connectionString = "<connection_string>";
+std::string eventHubName = "<event_hub_name>";
+
+Azure::Messaging::EventHubs::EventDataBatchOptions batchOptions;
+batchOptions.PartitionId = "1";
+Azure::Messaging::EventHubs::EventDataBatch eventBatch(batchOptions);
+
+Azure::Messaging::EventHubs::Models::EventData message;
+message.Body.Data = {'H', 'e', 'l', 'l', 'o', '2'};
+
+eventBatch.AddMessage(message);
+
+Azure::Messaging::EventHubs::ProducerClientOptions producerOptions;
+producerOptions.Name = "sender-link";
+producerOptions.ApplicationID = "some";
+
+auto client = Azure::Messaging::EventHubs::ProducerClient(
+      connectionString, eventHubName, producerOptions);
+auto result = client.SendEventDataBatch(eventBatch);
+```
+
+## Receive events
+
+The following example shows how to receive events from partition 1 on an event hub:
+
+```cpp
+// Your Event Hubs namespace connection string is available in the Azure portal.
+std::string connectionString = "<connection_string>";
+std::string eventHubName = "<event_hub_name>";
+
+auto client = Azure::Messaging::EventHubs::ConsumerClient(
+	connectionString, eventHubName);
+
+Azure::Messaging::EventHubs::PartitionClientOptions partitionOptions;
+// We want to consume all messages from the earliest (the default behavior is to consume new events).
+partitionOptions.StartPosition.Inclusive = true;
+partitionOptions.StartPosition.Earliest = true;
```

Comment:
It isn't clear from the sample why I would set these values to true compared to the default behavior. Is `Earliest` true by default? That's what the comment in brackets seems to suggest, to me.

Reply:
As the comment indicates, the default is to consume new events. This code says to include all events. But I pulled it out of this snippet.